crates/next-core/src/next_client_reference/visit_client_reference.rs

```diff
@@ -351,9 +358,16 @@ impl Visit<VisitClientReferenceNode> for VisitClientReference {
                         ty: VisitClientReferenceNodeType::ClientReference(
                             ClientReference {
                                 server_component: node.state.server_component(),
-                                ty: ClientReferenceType::EcmascriptClientReference(
-                                    client_reference_module,
-                                ),
+                                ty: ClientReferenceType::EcmascriptClientReference {
+                                    parent_module: Vc::try_resolve_downcast_type::<
+                                        EcmascriptClientReferenceProxyModule,
+                                    >(
+                                        parent_module
+                                    )
+                                    .await?
+                                    .unwrap(),
```

Comment:
In reality/at the moment, `EcmascriptClientReferenceProxyModule` is the only module which references `EcmascriptClientReferenceModule`. But this unwrap is not ideal of course.